### ISSUE TITLE
SNOW-986951: Python 3.12 - DeprecationWarning: datetime.datetime.utcfromtimestamp() is deprecated and scheduled for removal in a future version.

### ISSUE THREAD

**Issue by jlucas91** [snowflakedb/snowflake-connector-python#1824]
### Python version

3.12.0

### Operating system and processor architecture

Linux-5.10.47-linuxkit-x86_64-with-glibc2.36

### Installed packages

```python
N/A
```

### What did you do?

```python
Running the python Snowflake connector on python 3.12 produces multiple deprecation warnings.

  /usr/local/lib/python3.12/site-packages/snowflake/connector/converter.py:44: DeprecationWarning: datetime.datetime.utcfromtimestamp() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.fromtimestamp(timestamp, datetime.UTC).
    ZERO_EPOCH = datetime.utcfromtimestamp(0)

  /usr/local/lib/python3.12/site-packages/snowflake/connector/arrow_context.py:36: DeprecationWarning: datetime.datetime.utcfromtimestamp() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.fromtimestamp(timestamp, datetime.UTC).
    ZERO_EPOCH = datetime.utcfromtimestamp(0)
```

### What did you expect to see?

No deprecation warnings on python 3.12

### Can you set logging to DEBUG and collect the logs?

```bash
N/A
```

**Comment by sfc-gh-aling** [comment]
thanks for reaching out @jlucas91 
we will be working on Python 3.12 support and will make sure deprecations are handled properly.

**Comment by Koen-Vannisselroij** [comment]
What version of snoflake-python-connector are you using?
We had some trouble installing 3.6.0 on Python 3.12 and had to use Python 3.11 to get it to work.
Got an error related to 'C++ build tools' for 3.6.0
When trying 3.2.0 (this worked with another colleague) we got an error related to 'cmake'

The install for 3.6.0 worked with python 3.11 without issues.

**Comment by jlucas91** [comment]
Thanks @sfc-gh-aling! Do you have a rough sense of timeline? Timely 3.12 support from large providers in the ecosystem goes a long way to helping everyone move forward with a timely upgrade. 

We're on 3.5.0 @Koen-Vannisselroij . The build time dependencies snowflake requires seems to have changed (at least for us) over the months/years. You may need `libc6-dev` and `g++` installed.

**Comment by sfc-gh-aling** [comment]
hi folks, we have released python 3.12 on Jan 25th, cleaning up all the deprecation warnings.
I'm closing the issue now. please feel free to reach out if you still have questions.
